recipes/umbase/conanfile.py

```diff
@@ -0,0 +1,136 @@
+import sys
+import os
+
+from io import StringIO
+from pathlib import Path
+from typing import Optional
+
+from conans import ConanFile, tools
+from conans.errors import ConanException
+
+
+class UMBaseConanfile(object):
+    """
+    Ultimaker base conanfile, for reusing Python code in our repositories
+    https://docs.conan.io/en/latest/extending/python_requires.html
+    """
+
+    def _um_data(self, recipe_version: Optional[str]) -> dict:
+        """
+        Extract the version specific data out of a conandata.yml
+        """
+
+        if recipe_version:
+            if recipe_version in self.conan_data:
+                return self.conan_data[recipe_version]
+
+            recipe_version = tools.Version(recipe_version)
+            all_versions = []
+            for k in self.conan_data:
+                try:
+                    v = tools.Version(k)
+                except ConanException:
+                    continue
+                all_versions.append(v)
+            satifying_versions = sorted([v for v in all_versions if v <= recipe_version])
+            if len(satifying_versions) == 0:
+                raise ConanException(f"Could not find a maximum satisfying version for {recipe_version} in {[str(v) for v in all_versions]}")
+            version = str(satifying_versions[-1])
```

Comment:
```suggestion
            satifying_versions = [v for v in all_versions if v <= recipe_version]
            if len(satifying_versions) == 0:
                raise ConanException(f"Could not find a maximum satisfying version for {recipe_version} in {[str(v) for v in all_versions]}")
            version = str(max(satifying_versions))
```

I think something like this makes your intention a bit clearer

Reply:
I'm not sure if `tools.Version` plays well with python `max()` should be since I uses also use `sorted` and that also uses the comparators, so it should to work with `max`